src/System.Linq.Expressions/src/System/Runtime/CompilerServices/ReadOnlyCollectionBuilder.cs

```diff
@@ -170,12 +176,16 @@ public void RemoveAt(int index)
         {
             get
             {
-                ContractUtils.Requires(index < _size, nameof(index));
+                if (index >= _size)
```

Comment:
`List<T>` uses `uint` casts here to deal with the `< 0` check. As discussed in the description, not changing that in this PR because it's a breaking change from existing behavior (causing `IndexOutOfRangeException` because of indexing into the underlying array with a negative index). Will submit separate PR to suggest such changes.